File: C++/easy/217_contains_duplicate.cc
#include <iostream>
#include <vector>
#include <map>

using namespace std;

class Solution
{
public:
    bool containsDuplicate(vector<int> &nums)
    {
        unordered_map<int, int> m;
        for (size_t i : nums)
        {
            m[i]++;
            if (m[i] == 2)
                return true;
        }

        return false;
    }
};

File: Python/easy/217_contains_duplicate.py
from typing import List


class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        counts = {}
        for value in nums:
            curr = counts.get(value)
            if not curr:
                counts[value] = 1

            if curr == 2:
                return True

            counts[value] += 1

        return False
